cmd/ebrelayer/main.go

```diff
@@ -0,0 +1,168 @@
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdkContext "github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	"github.com/cosmos/cosmos-sdk/client/rpc"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtxb "github.com/cosmos/cosmos-sdk/x/auth/client/txbuilder"
+	amino "github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/libs/cli"
+
+	app "github.com/cosmos/peggy"
+	relayer "github.com/cosmos/peggy/cmd/ebrelayer/relayer"
+)
+
+const (
+	routeEthbridge = "ethbridge"
+)
+
+var defaultCLIHome = os.ExpandEnv("$HOME/.ebcli")
+var appCodec *amino.Codec
+
+func init() {
+
+	// Read in the configuration file for the sdk
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount(sdk.Bech32PrefixAccAddr, sdk.Bech32PrefixAccPub)
+	config.SetBech32PrefixForValidator(sdk.Bech32PrefixValAddr, sdk.Bech32PrefixValPub)
+	config.SetBech32PrefixForConsensusNode(sdk.Bech32PrefixConsAddr, sdk.Bech32PrefixConsPub)
+	config.Seal()
+
+	cdc := app.MakeCodec()
+	appCodec = cdc
+
+	// Add --chain-id to persistent flags and mark it required
+	rootCmd.PersistentFlags().String(client.FlagChainID, "", "Chain ID of tendermint node")
+	rootCmd.PersistentPreRunE = func(_ *cobra.Command, _ []string) error {
+		return initConfig(rootCmd)
+	}
+
+	// Construct Root Command
+	rootCmd.AddCommand(
+		rpc.StatusCommand(),
+		initRelayerCmd(),
+	)
+
+	executor := cli.PrepareMainCmd(rootCmd, "EBRELAYER", defaultCLIHome)
+	err := executor.Execute()
+	if err != nil {
+		panic(err)
+	}
+}
+
+var rootCmd = &cobra.Command{
+	Use:          "ebrelayer",
+	Short:        "Relayer service which listens for and relays ethereum smart contract events",
+	SilenceUsage: true,
+}
+
+//	initRelayerCmd : Initializes a relayer service run by individual
+//		validators which streams live events from a smart contract.
+//		The service automatically signs messages containing the event
+//		data and relays them to tendermint for handling by the
+//		EthBridge module.
+//
+func initRelayerCmd() *cobra.Command {
+	initRelayerCmd := &cobra.Command{
+		Use:   "init [web3Provider] [contractAddress] [eventSignature] [validatorFromName] --chain-id [chain-id]",
+		Short: "Initalizes a web socket which streams live events from a smart contract",
+		Args:  cobra.ExactArgs(4),
+		// NOTE: Preface both parentheses in the event signature with a '\'
+		Example: "ebrelayer init wss://ropsten.infura.io/ws ec6df30846baab06fce9b1721608853193913c19 LogLock(bytes32,address,bytes,address,uint256,uint256) validator --chain-id=testing",
+		RunE:    RunRelayerCmd,
+	}
+
+	return initRelayerCmd
+}
+
+//	RunRelayerCmd : RunRelayerCmd executes the initRelayerCmd with the provided parameters
+func RunRelayerCmd(cmd *cobra.Command, args []string) error {
+	// Parse chain's ID
+	chainID := viper.GetString(client.FlagChainID)
+	if chainID == "" {
```

Comment:
I think this needs `strings.TrimSpace`

Reply:
Fixed https://github.com/musnit/peggy/pull/6/commits/1b283ed11431bcdee13f9ec44dbd7356c7627e1a